src/filter/target.rs

```diff
@@ -21,25 +21,37 @@ use crate::filter::FilterResult;
 
 /// A filter that checks if the log level is higher than the specified level for a specific
 /// target.
-///
-/// Only if the target has a prefix that matches the target of the log record, the filter
-/// will be applied.
 #[derive(Debug, Clone)]
 pub struct TargetFilter {
     target: Cow<'static, str>,
     level: log::LevelFilter,
+    not: bool,
 }
 
 impl TargetFilter {
+    /// The filter will be applied only if the target **has** a prefix that matches the target of
+    /// the log record.
     pub fn level_for(target: impl Into<Cow<'static, str>>, level: log::LevelFilter) -> Self {
         TargetFilter {
             target: target.into(),
             level,
+            not: false,
+        }
+    }
+
+    /// The filter will be applied only if the target **does not have** a prefix that matches the
+    /// target of the log record,
+    pub fn level_for_not(target: impl Into<Cow<'static, str>>, level: log::LevelFilter) -> Self {
```

Comment:
Read challenging. IIRC it could be generally `neg_level_for` or `negtive_level_for`?

Reply:
It's not negative on `level` but negative on `for`. So I'd like to put not word after `for`.

Reply:
Maybe level_for_excluding?

Reply:
In practice, it is like:
```rust
let dispatch = Dispatch::new()
            .filter(TargetFilter::level_for_not(
                "databend::log::structlog",
                LevelFilter::Off,
            ))
            .append(structlog_log_file);
        logger = logger.dispatch(dispatch);
```

Reply:
I'd prefer to:

```rust
pub fn negative(mut self, negative: bool) -> Self {
  self.negative = negative;
  self
}
```

Maybe use `not`, I'm OK with it. But the pattern is to invert the matching result manually in a fluent method. What do you think?